debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/SharedClassesMetaDataHelper.java

```diff
@@ -23,14 +23,20 @@
 
 import com.ibm.j9ddr.CorruptDataException;
 import com.ibm.j9ddr.vm29.pointer.generated.J9ShrOffsetPointer;
+import com.ibm.j9ddr.vm29.types.UDATA;
+import java.lang.reflect.*;
 
 public class SharedClassesMetaDataHelper {
 	public static int getCacheLayerFromJ9shrOffset(J9ShrOffsetPointer j9shrOffset) throws CorruptDataException {
 		int layer;
 		try {
-			layer = j9shrOffset.cacheLayer().intValue();
-		} catch (NoSuchFieldError | NoSuchMethodError e) {
+			Method m = j9shrOffset.getClass().getMethod("cacheLayer");
+			UDATA layerUDATA = (UDATA) m.invoke(j9shrOffset);
+			layer = layerUDATA.intValue();
+		} catch (NoSuchMethodException | InvocationTargetException e) {
 			layer = 0;
+		} catch (Throwable e) {
```

Comment:
It's generally a bad idea to catch `Throwable`: what exceptions are anticipated?

Reply:
Removed this block. I expect `NoSuchMethodException, InvocationTargetException`. But compiler complains about `IllegalAccessException` from `Method.invoke()`, so I've added IllegalAccessException into the exception list in the previous block.